src/modules/form/components/DynamicFormField.tsx

```diff
@@ -0,0 +1,292 @@
+ * autofill behavior, and conditional rendering
+ */
+const DynamicFormField: React.FC<Props> = ({
+  handlers,
+  clientId,
+  warnIfEmpty = false,
+  horizontal = false,
+  visible = true,
+  pickListArgs,
+  item,
+  nestingLevel,
+  props: fieldProps,
+  renderFn,
+}) => {
+  const values = handlers.getValues();
+  const { definition, localConstants, getFieldErrors } = handlers;
+
+  // Handles changes to individual form items, converting values as needed
+  const itemChanged: ItemChangedFn = useCallback(
+    ({ linkId, value: baseValue, type }) => {
+      const item = handlers.itemMap[linkId];
+      let value = baseValue;
+      // Convert string values to appropriate number types
+      if (item) {
+        if (item.type === ItemType.Integer) value = parseInt(value) || 0;
+        if (item.type === ItemType.Currency) value = parseFloat(value) || 0;
+      }
+      handlers.methods.setValue(getSafeLinkId(linkId), value, {
+        shouldDirty: type === ChangeType.User,
+      });
+    },
+    [handlers]
+  );
+
+  // Handles batch changes to multiple form items
+  const severalItemsChanged: SeveralItemsChangedFn = useCallback(
+    ({ values, type }) => {
+      Object.entries(values).forEach(([linkId, value]) =>
+        itemChanged({ linkId, value, type })
+      );
+    },
+    [itemChanged]
+  );
+
+  // Renders the appropriate field component based on item type and configuration
+  const renderChild = useCallback(
+    (isDisabled?: boolean) => {
+      if (item.hidden) return null;
+
+      if (item.type === ItemType.Group) {
+        const group = (
+          <DynamicGroup
+            item={item}
+            clientId={clientId}
+            key={item.linkId}
+            nestingLevel={nestingLevel}
+            renderChildItem={(item, props, fn) => (
+              <DynamicFormField
+                key={item.linkId}
+                handlers={handlers}
+                item={item}
+                nestingLevel={nestingLevel + 1}
+                warnIfEmpty={warnIfEmpty}
+                pickListArgs={pickListArgs}
+                props={props}
+                renderFn={fn}
+              />
+            )}
+            renderSummaryItem={(item, isCurrency) => {
+              if (!item) return null;
+              return (
+                <AutofillFormItemWrapper handlers={handlers} item={item}>
+                  {(autofillValue) => {
+                    return isCurrency
+                      ? formatCurrency(autofillValue || 0)
+                      : autofillValue || 0;
+                  }}
+                </AutofillFormItemWrapper>
+              );
+            }}
+            values={values}
+            itemChanged={itemChanged}
+            severalItemsChanged={severalItemsChanged}
+            visible={visible}
+            debug={
+              import.meta.env.MODE === 'development'
+                ? (keys?: string[]) => {
+                    const sectionValues = keys ? pick(values, keys) : values;
```

Comment:
An adjustment I made locally so I could use the "Debug" button on the form during local development. `values` is stale here otherwise
```suggestion
                    const currentValues = handlers.getValues();
                    const sectionValues = keys
                      ? pick(currentValues, keys)
                      : currentValues;
```